src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Data
+{
+    public sealed partial class TextLoaderRange
```

Comment:
Having `DebuggerDisplay` showing the ranges could be helpful here.

Reply:
Can you please clarify the debugging scenario you are thinking about? If you are in the TextLoader you can inspect ranges by looking into the column class.

---
In reply to: [188839042](https://github.com/dotnet/machinelearning/pull/142#discussion_r188839042) [](ancestors = 188839042)

Reply:
I'm not sure if I understand the question. I'm thinking straight that the difference is this:
![DebuggerDisplay](https://user-images.githubusercontent.com/5158694/40296490-64675ed6-5ce5-11e8-8767-95f142dbccc3.png)

Where having the attribute is more pleasing. It also sets the tone of the codebase that one should apply aids where appropriate. So to speak, not to set a tone in which to leave things out if one can get away it by "doing something else", but be helpful on the spot.

Reply:
I actually rather like this idea. `"Min = {Min}, Max = `{Max}`". Maybe also utilize the underlying method `TryUnparse` in some fashion in the corresponding `Runtime` class, but is a bit more idiosyncratic.

---
In reply to: [189518236](https://github.com/dotnet/machinelearning/pull/142#discussion_r189518236) [](ancestors = 189518236)

Reply:
While I was driving in I was thinking about this, and I like the idea, but I think we could possibly do something a bit more comprehensive than a one-off here. I have however added a new issue #194. where I outline an idea where perhaps we could actually generate this attribute for literally *everything* in `CSharpApi.cs`. It might actually be really easy to do... (That of course would be beyond the scope of this change.)

---
In reply to: [189653741](https://github.com/dotnet/machinelearning/pull/142#discussion_r189653741) [](ancestors = 189653741,189518236)

Reply:
Thanks, Veikko and Tom. I think this is a great suggestion but unfortunately out of scope for this change. 

---
In reply to: [189661430](https://github.com/dotnet/machinelearning/pull/142#discussion_r189661430) [](ancestors = 189661430,189653741,189518236)